protocol/streaming/full_node_streaming_manager.go

```diff
@@ -29,9 +29,12 @@ type FullNodeStreamingManagerImpl struct {
 	ticker *time.Ticker
 	done   chan bool
 
-	// map of clob pair id to stream updates.
-	streamUpdateCache map[uint32][]clobtypes.StreamUpdate
-	numUpdatesInCache uint32
+	// list of stream updates.
+	streamUpdateCache []clobtypes.StreamUpdate
+	// list of subscription ids for each stream update.
+	streamUpdateSubscriptionCache     [][]uint32
+	clobPairIdToSubscriptionIdMapping map[uint32][]uint32
+	numUpdatesInCache                 uint32
```

Comment:
prob no need for this field now that we have a central list, can use len(list)

Reply:
removed